http/client.go

```diff
@@ -90,3 +95,126 @@ func (c *traceClient) Do(r *http.Request) (*http.Response, error) {
 	tracing.InjectToHTTPRequest(span, r)
 	return c.Client.Do(r)
 }
+
+// C is a basic http client that can make cReqs with out having to juggle
+// the token and so forth. It provides sane defaults for checking response
+// statuses, sets auth token when provided, and sets the content type to
+// application/json for each request. The token, response checker, and
+// content type can be overidden on the cReq as well.
+type C struct {
```

Comment:
Just a note that this struct name could be confusing and without context could be misinterpreted to reference CGO

Reply:
great point, totally forgot to comeback and touch that up. Time to do the hardest thing in CS, name something 🤔